seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SinkFlowLifeCycle.java

```diff
@@ -354,10 +343,5 @@ public void restoreState(List<ActionSubtaskState> actionStateList) throws Except
         } else {
             this.writer = sinkAction.getSink().restoreWriter(writerContext, states);
         }
-        if (this.writer instanceof SupportResourceShare) {
-            resourceManager =
-                    ((SupportResourceShare) this.writer).initMultiTableResourceManager(1, 1);
-            ((SupportResourceShare) this.writer).setMultiTableResourceManager(resourceManager, 0);
```

Comment:
This can trigger the method org.apache.seatunnel.connectors.seatunnel.jdbc.sink.JdbcSinkWriter#setMultiTableResourceManager which is in jdbcsink that can create a SimpleJdbcConnectionPoolProviderProxy.

Reply:
It be trigger in https://github.com/apache/seatunnel/blob/dev/seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/multitablesink/MultiTableSinkWriter.java#L117

Reply:
Ok